pkg/registry/maven/maven.go

```diff
@@ -173,3 +174,22 @@ func (r HTTPRegistry) ReleaseURL(ctx context.Context, pkg, version, typ string)
 	artifactURL := releaseURL.JoinPath(artifactPath)
 	return artifactURL.String(), nil
 }
+
+// Artifact returns file that is part of the Maven release.
+func (r HTTPRegistry) Artifact(ctx context.Context, pkg, version, artifact string) (io.ReadCloser, error) {
+	g, a, found := strings.Cut(pkg, ":")
+	if !found {
+		return nil, errors.New("package identifier not of form 'group:artifact'")
+	}
+	artifactPath := path.Join(strings.ReplaceAll(g, ".", "/"), a, version, artifact)
+	artifactURL := releaseURL.JoinPath(artifactPath)
+	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, artifactURL.String(), nil)
```

Comment:
hmm we might want to check this error, just to be sure. it's unlikely to be a concern but there are some characters in `g` that could cause the URL to be invalid (however unlikely that might be).